src/codegen.c

```diff
@@ -468,10 +468,11 @@ void code_generate()
             if (dump_ir == 1)
                 printf("    x%d >>= x%d", dest_reg, OP_reg);
             break;
-        case OP_not:
+        case OP_log_not:
             /* 1 if zero, 0 if nonzero */
-            /* only works for 1/0 */
-            emit(__rsb_i(__AL, dest_reg, 1, dest_reg));
+            emit(__teq(dest_reg));
```

Comment:
Is it feasible to avoid the setting for both `0` and `1` every time we counter logical not?

Reply:
Yes it is feasible, but code length must be 12. Code is update~